src/UnitTests/Substitutes.cs

```diff
@@ -31,7 +32,17 @@ internal static class Substitutes
 
 
        // public static IGitRepositoriesExt IGitRepositoriesExt { get { return Substitute.For<IGitRepositoriesExt>(); } }
-        public static IGitService IGitService { get { return Substitute.For<IGitService>(); } }
+        public static IGitService IGitService
+        {
+            get
+            {
+                var result = Substitute.For<IGitService>();
+                var repo = Substitute.For<IRepository>();
+                repo.Head.Returns(Substitute.For<Branch>());
+                result.GetRepo(Arg.Any<string>()).Returns(repo);
+                return result;
```

Comment:
This shouldn't be done here, these are assumptions that are going to be different depending on what we're testing (what if we want to test the behaviour when there's no Head? what if we want to have different things returned when we call `GetRepo`?). The test should be setting all of this up by itself.

The reason we have these mocked service properties is so that they can be singletons to pass to constructors and return via the service provider. They're services that always exist but are empty of any specific data, and tests customize the return values of their properties and methods depending on the situation.

All test classes derive from TestBaseClass, which has Entry and Exit virtual methods that are called by every test method at the beginning and end. If you want to do common setup for every test in a class, you can override them and set things up there.